src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,560 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import errno
+from logging import getLogger
+import os
+import re
+
+from ClusterShell.NodeSet import NodeSet
+
+from exception_utils import CommandFailure
+from general_utils import run_task, display_task
+
+SUPPORTED_PROVIDERS = (
+    "ofi+sockets", "ofi+tcp", "ofi+tcp;ofi_rxm", "ofi+verbs", "ofi+verbs;ofi_rxm", "ucx+dc_x")
+
+
+class NetworkDevice():
+    """A class to represent the information of a network device."""
+
+    def __init__(self, host, device, ib_device, port, provider, numa):
+        """Initialize the network device data object."""
+        self.host = host
+        self.device = device
+        self.ib_device = ib_device
+        self.port = port
+        self.provider = provider
+        self.numa = numa
+
+    def __repr__(self):
+        """Overwrite to display formatted devices."""
+        return self.__str__()
+
+    def __str__(self):
+        """Overwrite to display formatted devices."""
+        settings = ["{}={}".format(key, getattr(self, key, None)) for key in self.__dict__]
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string